### PR TITLE
Alter Issue Number Extraction RegEx from Branch Name

### LINKED ISSUE #103: Alter and/or document hook logic
I've started using the git commit hooks provided by RDM and ran into an unexpected outcome today. I'm not sure I would classify it as a bug, but I think it might not match assumptions made by users and should probably either be updated or perhaps even just documented with a warning somewhere.

> (I also noticed that `rdm hooks` is documented within `init_files`, but not in the main README. I've already found the hook useful after just a few days of use, so it seems to me worth highlighting in the main README, perhaps in the *Quick Start* section?)

## Issue

Given a branch name of `245-fix-uploads-to-s3`, the commit message will be pre-populated with:

```

# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch 245-fix-uploads-to-s3
# Changes to be committed:
#	...
#

Issue #245

Issue #3
```

I certainly didn't intend for `Issue #3` to get linked, nor do I think most users would.

I suppose this probably doesn't happen that often, but could still see it happening more than once, with branch names like `__-migrate-to-python-v3`, `__-implement-s3`, `__-fix-base64-usage`, etc.

## Cause

Looks like the pattern for extracting the issue[ is pretty broad (`[0-9]+`)](https://github.com/innolitics/rdm/blob/e2ab8b02d42b003a7064d5c460457cff3bff0f8e/rdm/hook_files/prepare-commit-msg#L8).

## Proposed solution
Although users can easily edit the script themselves, it might be nice to either call this out somewhere in the docs, or refine the regex a little to provide a different default.

For example, although not very nice to look at (this was a quick pass; could probably be refined more), this avoids the above issue, as well as some other edge-cases:

```diff
- ISSUES=$(echo "$BRANCH" | grep -o -E '[0-9]+')
+ ISSUES=$(echo "$BRANCH" | grep -o -E '^([0-9]+)|[^A-z.0-9]([0-9]+)[^A-z.0-9]*?' | grep -o -E '[0-9]+')
```

Test cases [here](https://regexr.com/6j35l).

### PR DESCRIPTION
**Pull request by joshuatz**
Closes #103

In addition to altering the issue number RegEx and adding notes to the README on `rdm hooks`, I also threw in a quick typo fix and adding a language specifier to the MD code fence block in `Quick Install`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -319,6 +325,16 @@ To provide a custom checklist, use a file path for the first argument.
...
 
+## Git Hooks
+
+The `rdm hooks` command installs git hooks that will automatically add change request numbers to your git commit messages, based on your branch name.
+
+> When using this feature, name your branches in the style of `${hyphen-separated-issue-numbers}-rest-of-branch-name`.
+>
+> Example: `12-13-fixing-issues`
+
+If you manually edit one of the provided hooks and want to revert your changes or RDM has been updated and you want to make sure the newest hooks are used, you can re-run this command at any time.
```
Comment (joshuatz) [inline README.md:336]: Side-note: I noticed that we "install" our git hooks by copying the files over (via `copy_directory` util, which is then using `shutil.copy2`) - thus the reasoning for this note.

Any reason why we are not using symlinks? I think that would let us avoid upgrade issues, where the user has to manually re-run this install command in order to get upgraded hooks after pip upgrading.

File: README.md [hunk 2]
```diff
@@ -19,8 +19,13 @@ RDM is especially well-suited for early-stage software-only medical devices that
 
 ## Quick Start
 
-```
+```sh
```
Comment (johndgiese) [inline README.md:22]: 👍

**Review by johndgiese — COMMENTED** [review]
Nice work @joshuatz , I really appreciate you taking the initiative to bring this up and then to execute on pushing up a fix! I had a few small suggested changes, but it's pretty much good to merge.

File: README.md [hunk 3]
```diff
@@ -319,6 +325,16 @@ To provide a custom checklist, use a file path for the first argument.
...
 The checklist format is described in detail [here](./docs/checklist-format.md).
 
+## Git Hooks
+
+The `rdm hooks` command installs git hooks that will automatically add change request numbers to your git commit messages, based on your branch name.
+
+> When using this feature, name your branches in the style of `${hyphen-separated-issue-numbers}-rest-of-branch-name`.
+>
+> Example: `12-13-fixing-issues`
+
```
[the remark this answers is not in the document]
Comment (joshuatz) [reply README.md:337]: Added in 901a8df9253318347e763f1a94cc7863950837a9.
